### PR TITLE
Enable `prefer-destructuring` rule

### PR DESCRIPTION
**Pull request by twokul**

### CODE REVIEW

File: lib/commands/destroy.js [hunk 1]
```diff
@@ -28,8 +28,8 @@ module.exports = Command.extend({
   beforeRun: mergeBlueprintOptions,
 
   run(commandOptions, rawArgs) {
-    let blueprintName = rawArgs[0];
-    let entityName = rawArgs[1];
+    let [blueprintName] = rawArgs;
+    let [, entityName] = rawArgs;
```
Comment (kellyselden) [inline lib/commands/destroy.js]: looks like the autofixer screwed up here

File: lib/commands/destroy.js [hunk 2]
```diff
@@ -61,7 +61,7 @@ module.exports = Command.extend({
   },
 
   printDetailedHelp() {
-    let ui = this.ui;
+    const { ui } = this;
```
Comment (kellyselden) [inline lib/commands/destroy.js]: keep as let

File: lib/errors/silent.js [hunk 3]
```diff
@@ -6,9 +6,11 @@ const deprecate = require('../utilities/deprecate');
 Object.defineProperty(module, 'exports', {
   get() {
     // Get the call stack so we can let the user know what module is using the deprecated function.
+    /* eslint-disable prefer-destructuring */
     let stack = new Error().stack;
```
Comment (kellyselden) [inline lib/errors/silent.js]: `let { stack } = new Error();` won't work here?

File: lib/models/command.js [hunk 4]
```diff
@@ -459,7 +459,7 @@ let Command = CoreObject.extend({
         key = alias;
         value = [`--${option.name}`];
       } else if (aliasType === 'object') {
-        key = Object.keys(alias)[0];
+        [key] = Object.keys(alias);
```
Comment (kellyselden) [inline lib/models/command.js]: it's interesting this is allowed but

```
    // eslint-disable-next-line prefer-destructuring
     version = require(path.join(emberCLIBabelInstance.root, 'package')).version;
```

above isn't. hmm.

File: lib/tasks/server/livereload-server.js [hunk 5]
```diff
@@ -21,12 +21,12 @@ function createServer(options) {
...
 
 function isNotRemoved(entryTuple) {
-  let operation = entryTuple[0];
+  let [operation] = entryTuple;
   return operation !== 'unlink' && operation !== 'rmdir';
 }
 
 function isNotDirectory(entryTuple) {
-  let entry = entryTuple[2];
+  let [, , entry] = entryTuple;
```
Comment (kellyselden) [inline lib/tasks/server/livereload-server.js]: that's a little unfortunate.

**Comment by Turbo87** [comment]
I'd prefer not to use this rule as it can lead to much less readable code in some instances as @kellyselden has already pointed out. I'm okay with using destructuring if it improves readability, but I'm against enforcing it.

**Comment by twokul** [comment]
@Turbo87 I generally don't like inconsistency throughout (we have a fair share of it in our codebase) and I would prefer a mix of enforce + `eslint-disable` when needed.

That being said I can close if that's not something we want to do 👩‍💻

**Comment by Turbo87** [comment]
I would personally vote for closing, but let's wait what the others have to say.

File: lib/tasks/server/middleware/broccoli-serve-files/package.json [hunk 6]
```diff
@@ -3,6 +3,9 @@
   "keywords": [
     "ember-addon"
   ],
+  "engines": {
```
Comment (kellyselden) [inline lib/tasks/server/middleware/broccoli-serve-files/package.json]: looks like these snuck in

**Comment by twokul** [comment]
@ember-cli/core thoughts?

**Comment by Turbo87** [comment]
closing due to inactivity

**Pull request closed without merging**